### ISSUE TITLE
Extract Interface creates private classes

### ISSUE THREAD

**Issue by retailcoder** [rubberduck-vba/Rubberduck#5346]
### Bug

If the interface is being extracted from a `PublicNotCreatable` class, the generated interface must also be `PublicNotCreatable`, otherwise the refactoring causes a compile error due to inconsistent accessibility (public class cannot implement a private interface).

### Enhancement

It would be nice to add an "Instancing" combobox to the refactoring UI, or "Private" and "Public (not creatable)" optionbuttons, with "Public" being the default, and "Private" being only enabled if the source class is also private; that way the refactoring could extract a public or private interface class out of a private class, or a public interface out of a public class.

This is also an opportunity to review the annotations we're generating the interface with: the interface should have an `@Interface` annotation, and if public, an `@Exposed` annotation as well.
